lib/stores/FileStore.ts

```diff
@@ -1,45 +1,55 @@
+// TODO: use /promises versions
 import fs from 'node:fs'
 import path from 'node:path'
 import stream from 'node:stream'
 
 import debug from 'debug'
-
 import Configstore from 'configstore'
+
 import DataStore from './DataStore'
 import pkg from '../../package.json'
 import {ERRORS} from '../constants'
 
+import type {File} from '../../types'
+
+type Store = {
+  get(key: string): File | undefined
+  set(key: string, value: File): void
+  delete(key: string): void
+}
+
+type Options = {
+  directory: string
+  configstore?: Store
+}
+
 const MASK = '0777'
 const IGNORED_MKDIR_ERROR = 'EEXIST'
 const FILE_DOESNT_EXIST = 'ENOENT'
 const log = debug('tus-node-server:stores:filestore')
 
-class FileStore extends DataStore {
+export default class FileStore extends DataStore {
   directory: string
-  configstore: Configstore
-
-  constructor(options: any) {
-    super(options)
-    this.directory = options.directory
-    this.configstore = options.configstore
-
-    if (!this.configstore) {
-      this.configstore = new Configstore(`${pkg.name}-${pkg.version}`)
-    }
+  configstore: Store
 
+  constructor({directory, configstore}: Options) {
+    super()
+    this.directory = directory
+    this.configstore = configstore ?? new Configstore(`${pkg.name}-${pkg.version}`)
     this.extensions = [
       'creation',
       'creation-with-upload',
       'creation-defer-length',
       'termination',
     ]
-    this._checkOrCreateDirectory()
+    // TODO: this async call can not happen in the constructor
+    this.checkOrCreateDirectory()
   }
 
   /**
    *  Ensure the directory exists.
    */
-  _checkOrCreateDirectory() {
+  private checkOrCreateDirectory() {
     fs.mkdir(this.directory, MASK, (error) => {
       if (error && error.code !== IGNORED_MKDIR_ERROR) {
         throw error
```

Comment:
I agree with the comment in the constructor, that throwing an error which cannot be handled at all it not the best approach :)

Reply:
Yes, we're handling this in a separate issue (#300) and PR.